exopy/utils/widgets/qt_tree_widget.py

```diff
@@ -766,17 +769,13 @@ def _drop_object(self, node, obj, dropped_object, make_copy=True):
 
         return copy.deepcopy(new_object)
 
-    def _on_nid_changed(self, nid, col):
+    def _on_data_committed(self, editor):
         """Handle changes to a widget item subsequent to a renaming operation.
 
         """
-        # The node data may not have been set up for the nid yet.  Ignore it if
-        # it hasn't.
-        try:
-            _, node, obj = self._get_node_data(nid)
-        except Exception:
-            return
-
+        nid = self.get_widget().currentItem()
+        col = self.get_widget().currentColumn()
+        _, node, obj = self._get_node_data(nid)
```

Comment:
Are you sure the try except is not needed anymore ?

Reply:
I've tested it for a few minutes and it worked.
Moreover, `commitData` is a lot more selective that `itemChanged` and is only called when the user finishes editing an item so I don't think this signal can be emitted before the data has been set up.

Reply:
Ok. I will merge tomorrow then *or later tonight).